### PR TITLE
aether-38: extend jsonpath_ng with custom wildcards

### PR DESCRIPTION
**Pull request by quesebifurcan**
Resolves https://jira.ehealthafrica.org/browse/AET-38

TODO:
 - [x] add a predicate function `is_custom_jsonpath` + some more tests.

### CODE REVIEW

File: gather2-core/gather2/core/api/utils.py [hunk 1]
```diff
@@ -58,6 +60,48 @@ def json_printable(obj):
         return obj
 
 
+custom_jsonpath_wildcard_regex = re.compile('[a-zA-Z0-9_-]+\*')
```
Comment (admbtlr) [inline gather2-core/gather2/core/api/utils.py]: Do we need a `.` in there to allow for more nested json paths, e.g. `$.facility.dose-*.id`?

**Review by admbtlr — CHANGES_REQUESTED** [review]

File: gather2-core/gather2/core/api/utils.py (hunk 1 above)
Comment (quesebifurcan) [reply gather2-core/gather2/core/api/utils.py]: Good point -- @shawnsarwar said yesterday that aet-38 should only handle "top-level" custom wildcards. Which might be a good call -- the nested implementation would be significantly more complicated.

Comment (quesebifurcan) [reply gather2-core/gather2/core/api/utils.py]: Come to think of it, I should probably add some more tests to make sure that we always fall back to `jsonpath_ng.parse` when we should. Here's a good list: https://github.com/json-path/JsonPath#path-examples

Comment (admbtlr) [reply gather2-core/gather2/core/api/utils.py]: Yup, good call on the tests. And fair enough about ignoring nesting for now.

**Review by admbtlr — APPROVED** [review]

**Pull request merged**